### DISCUSSION TITLE
How can I use higher fidelity model in JSBSim?

### DISCUSSION THREAD

**Discussion by DreamerSurpass** [JSBSim-Team/jsbsim#994] [Q&A]
I noticed that the F-16 xml configuration file used low fidelity aerodynamic data(alpha range is -10 to 45 degree). But in NASA TP-1538 wind tunnel data alpha range is -20 to 90 degree. Is it possible to use higher fidelity model in JSBSim?

**Comment by seanmcleod** [comment]
Sure, supply higher fidelity data and JSBSim will use it.

**Reply by DreamerSurpass** [reply]
Thanks for your quick reply.
So do I have to manually implement the hight fidelity data F-16 file? This seems like a little bit of work. And if I replace the aerodynamic part, what about other files( like engine)?
Or where can I download it conveniently? I have downloaded the F-16 model in FlightGear and found that it seems use high fidelity data. Is it possilble to use that model directly in JSBSim?

**Comment by seanmcleod** [comment]
> This seems like a little bit of work.

Someone has to do it :wink:

> And if I replace the aerodynamic part, what about other files( like engine)?

Different parts can have different levels of fidelity. But yes if you want high fidelity for everything then you have to source the required data for each component and if it isn't already in the JSBSim FDM data format then also convert it.

> I have downloaded the F-16 model in FlightGear and found that it seems use high fidelity data.

So you looked at the FDM data file and you saw a difference in the data?

> Is it possilble to use that model directly in JSBSim?

If it's a JSBSim based FDM then generally yes, although FlightGear does also support additional scripting via Nasal etc. so if a lot of that has been included then it will be more work to remove the Nasal specific code. But if you're just looking for the core FDM data that shouldn't be a big deal to use with JSBSim stand-alone.

**Reply by DreamerSurpass** [reply]
> So you looked at the FDM data file and you saw a difference in the data?

Yes，upper is the data file I found in FlightGear model and the picture below is the base f16 file in JSBSim. The upper one has a larger range.
![image](https://github.com/JSBSim-Team/jsbsim/assets/93421761/38fe7f83-ff5e-4398-b15e-34f9d1dcc05c)
![image](https://github.com/JSBSim-Team/jsbsim/assets/93421761/842bcc6b-ff0b-4296-a0a4-50b659c05145)

> But if you're just looking for the core FDM data that shouldn't be a big deal to use with JSBSim stand-alone.

Got it, thanks a lot.

**Comment by seanmcleod** [comment]
And I see they mention in a note who went to the effort of typing in all the numbers :wink:
